PWGGA/GammaConv/macros/AddTask_GammaConvV1_PbPb.C

```diff
@@ -4337,106 +4341,120 @@ void AddTask_GammaConvV1_PbPb(
       analysisEventCuts[i]->SetUseWeightMultiplicityFromFile(kTRUE, fileNameMultWeights, dataInputMultHisto, mcInputMultHisto );
     }
 
-    TString histoNameMCPi0PT = "";  TString histoNameMCEtaPT = "";  TString histoNameMCK0sPT = "";    // pT spectra to be weighted
-    TString fitNamePi0PT = "";      TString fitNameEtaPT = "";      TString fitNameK0sPT = "";        // fit to correct shape of pT spectra
-    Bool_t weightPi0 = kFALSE;      Bool_t weightEta = kFALSE;      Bool_t weightK0s = kFALSE;
-    if(enablePtWeighting){
-      cout << "INFO enabeling pT weighting" << endl;
-      if(periodNameAnchor.CompareTo("LHC15o")==0 || periodNameAnchor.CompareTo("LHC18q")==0){
-        TString eventCutString  = cuts.GetEventCut(i);
-        TString eventCutShort   = eventCutString(0,6);   // first six digits
-        weightPi0         = kTRUE;
-        histoNameMCPi0PT  = Form("Pi0_%s_5TeV_%s",   periodNameV0Reader.Data(), eventCutString.Data());  // MC
-        fitNamePi0PT      = Form("Pi0_Data_5TeV_%s", eventCutShort.Data());                              // fit to data
-        weightEta         = kTRUE;
-        histoNameMCEtaPT  = Form("Eta_%s_5TeV_%s",   periodNameV0Reader.Data(), eventCutString.Data());
-        fitNameEtaPT      = Form("Eta_Data_5TeV_%s", eventCutShort.Data());
+    TString histoNameMCPi0PT = "";
+    TString histoNameMCEtaPT = "";
+    TString histoNameMCK0sPT = ""; // pT spectra to be weighted
+    TString fitNamePi0PT = "";
+    TString fitNameEtaPT = "";
+    TString fitNameK0sPT = ""; // fit to correct shape of pT spectra
+    Bool_t weightPi0 = kFALSE;
+    Bool_t weightEta = kFALSE;
+    Bool_t weightK0s = kFALSE;
+    if (intPtWeightsCalculationMethod)
+    {
+      printf("AddTask_GammaConvV1_PbPb.C: INFO: intPtWeightsCalculationMethod = %d\n", intPtWeightsCalculationMethod);
+      if (periodNameAnchor.BeginsWith("LHC15o") || 
+          periodNameAnchor.BeginsWith("LHC18q"))
+      {
+        TString eventCutString = cuts.GetEventCut(i);
+        TString eventCutShort = eventCutString(0, 6);                                                // first six digits
+        histoNameMCPi0PT = Form("Pi0_%s_5TeV_%s", periodNameV0Reader.Data(), eventCutString.Data()); // MC
+        fitNamePi0PT = Form("Pi0_Data_5TeV_%s", eventCutShort.Data());                               // fit to data
+        histoNameMCEtaPT = Form("Eta_%s_5TeV_%s", periodNameV0Reader.Data(), eventCutString.Data());
+        fitNameEtaPT = Form("Eta_Data_5TeV_%s", eventCutShort.Data());
       }
-      analysisEventCuts[i]->SetUseReweightingWithHistogramFromFile(weightPi0, weightEta, weightK0s, fileNamePtWeights, histoNameMCPi0PT, histoNameMCEtaPT, histoNameMCK0sPT, fitNamePi0PT, fitNameEtaPT, fitNameK0sPT);
```

Comment:
Why were the different weights, like "weightPi0" changed to "intPtWeightsCalculationMethod" and the K0s was disabled completely?

Reply:
1) the semantics of  weightPi0 and alike do not change. Unless I oversee sth. Which change do you see?
2) I renamed the macro argument parameter 'enablePtWeighting' -> 'intPtWeightsCalculationMethod' because I also changed its type from bool to int, because Im using it steer if  invariantly or variantly calculated weights shall be used. I should have used a separate commit just for the renaming and type change of this variable to make this more clear, but forgot about it in the beginning. Sorry for that.
3) Note: 
 -  that the lines here are in the scope of 'if(periodNameAnchor.BeginsWith("LHC15o") || periodNameAnchor.BeginsWith("LHC18q"))'. Meike and I are not using any weights for K0s, thats why I don't switch them on.
 - that also in the current version of the code all the weights are initialised as off and then afterwards, depending on anchored data and mc tags switched on.